src/main/scala/akka/persistence/jdbc/journal/dao/JournalDaoWithUpdates.scala

```diff
@@ -0,0 +1,30 @@
+/*
+ * Copyright 2016 Dennis Vriend
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package akka.persistence.jdbc.journal.dao
+
+import scala.concurrent.Future
+
+/**
+ * A [[JournalDao]] with extended capabilities, such as updating payloads and tags of existing events.
+ * These operations should be used sparingly, for example for migrating data from un-encrypted to encrypted formats
+ */
+trait JournalDaoWithUpdates extends JournalDao {
```

Comment:
Do we really need it as a trait? Is the motivation behind it to not disturb users that are using their own implementation of `JournalDao`?

I personally don't think this should be a problem. If we add the method to `JournalDao`, users implementing their own DAO will get compiler error when upgrading. 

Unless they get it upgraded transitively, which is rather rare case, they will get a runtime error when using the method, but that is what they will get anyway with the current implementation.

Reply:
I do think the separation is clear, not sure conflating journal api with this mutable terrible api it is helping

Reply:
Fine by me. 

I just don't find it useful to separate it. As as consequence we need to branch later on (like here https://github.com/dnvriend/akka-persistence-jdbc/pull/183/files#diff-381cd8e4dd090538bdd312eed8222f76R75).

Anyway, you don't need to change it. Just want to understand the motivation in creating a separated trait.